x-pack/plugins/observability_solution/logs_explorer/public/state_machines/logs_explorer_controller/src/services/selection_service.ts

```diff
@@ -5,59 +5,103 @@
  * 2.0.
  */
 
+import { DiscoverStart } from '@kbn/discover-plugin/public';
 import { InvokeCreator } from 'xstate';
 import { Dataset } from '../../../../../common/datasets';
-import { SingleDatasetSelection } from '../../../../../common/dataset_selection';
+import {
+  isDataViewSelection,
+  isUnresolvedDatasetSelection,
+  SingleDatasetSelection,
+  UnresolvedDatasetSelection,
+} from '../../../../../common/dataset_selection';
 import { IDatasetsClient } from '../../../../services/datasets';
 import { LogsExplorerControllerContext, LogsExplorerControllerEvent } from '../types';
+import { redirectToDiscover } from './discover_service';
 
-interface LogsExplorerControllerUrlStateDependencies {
+interface LogsExplorerControllerSelectionServiceDeps {
   datasetsClient: IDatasetsClient;
+  discover: DiscoverStart;
 }
 
-export const validateSelection =
+export const initializeSelection =
   ({
     datasetsClient,
-  }: LogsExplorerControllerUrlStateDependencies): InvokeCreator<
+    discover,
+  }: LogsExplorerControllerSelectionServiceDeps): InvokeCreator<
     LogsExplorerControllerContext,
     LogsExplorerControllerEvent
   > =>
   (context) =>
   async (send) => {
-    const unresolvedIntegrationName =
-      context.datasetSelection.selection.dataset.parentIntegration?.name;
-    const unresolvedDatasetName = context.datasetSelection.selection.dataset.name;
+    /**
+     * First validation.
+     * If the selection is a data view which is not of logs type, redirect to Discover.
+     */
+    if (
+      isDataViewSelection(context.datasetSelection) &&
+      context.datasetSelection.selection.dataView.isUnknownDataType()
+    ) {
+      return redirectToDiscover({ context, datasetSelection: context.datasetSelection, discover });
```

Comment:
Similarly, should this side-effect happen here? Or should it be delegated to the consumer too?

Reply:
The same work mentioned [here](https://github.com/elastic/kibana/pull/176078#discussion_r1487731998) will apply to this part.